### ISSUE TITLE
Use system clipboard on MacOS

### ISSUE THREAD

**Issue by bscuron** [martanne/vis#1067]
I am currently trying to use the system clipboard by default, but I receive this message when I compile with the changes by @deepcube mentioned [here](https://github.com/martanne/vis/issues/1003#issuecomment-1118627247): "Command failed System clipboard not supported". I also tried just setting the primary register = clipboard register in `vis.h`. This works inside of vis, but when I paste outside of the editor it does not paste the contents of the yanked text.

Please let me know if you have any ideas that may help. Thank you.

**Comment by QiBaobin** [comment]
The `*` doesn't work, but `+` register works for me, you can use that instead.
```
  +   System clipboard register, see vis-clipboard(1)
  *   Primary clipboard register, see vis-clipboard(1)
```

**Comment by mcepl** [comment]
Yes, it is system dependent and Mac OS (and Windows) doesn’t have whole concept of selection, right? When you select text in one application and then middle-click in other one, nothing happens, right?

I think this ticket can be closed, because we can do nothing about your system.

**Comment by rnpnr** [comment]
I'm wondering if this behaviour even makes sense. Wouldn't it be better to just copy to the available clipboard and ignore the `--selection` argument if its not supported by the system?

**Comment by bscuron** [comment]
This worked for me on MacOS:
```lua
    vis.registers['"'] = { io.popen("vis-clipboard --paste"):read("*all") } -- use system clipboard
``` This was suggested by @leorosa in #1003
